src/Powercord/modules/jsx.js

```diff
@@ -17,14 +17,40 @@
  */
 
 const sucrase = require('sucrase');
-const { readFileSync } = require('fs');
+const { join } = require('path');
+const { readFileSync, existsSync, mkdirSync, writeFile } = require('fs');
+const { createHash } = require('crypto');
 
-module.exports = () =>
-  require.extensions['.jsx'] = (_module, filename) =>
-    _module._compile(
-      sucrase.transform(readFileSync(filename, 'utf8'), {
+const cacheDir = join(__dirname, '../../../cache/jsx/');
```

Comment:
gitignore mentions `.cache` but you use `cache`
```suggestion
const cacheDir = join(__dirname, '../../../.cache/jsx/');
```